web/html/src/manager/virtualization/guests/list/guests-list.js

```diff
@@ -1,406 +1,441 @@
-/* eslint-disable */
 // @flow
 
-const React = require("react");
-const PropTypes = React.PropTypes;
-const MessageContainer = require("components/messages").Messages;
-const MessagesUtils = require("components/messages").Utils;
-const {Table, Column, SearchField, Highlight} = require("components/table");
-const Functions = require("utils/functions");
-const Utils = Functions.Utils;
-const {LinkButton, AsyncButton} = require("components/buttons");
-const {DeleteDialog} = require("components/dialog/DeleteDialog");
-const {DangerDialog} = require("components/dialog/DangerDialog");
-const {ModalButton} = require("components/dialog/ModalButton");
-const Systems = require("components/systems");
-const {VirtualizationGuestActionApi} = require('../virtualization-guest-action-api');
-const {VirtualizationGuestsListRefreshApi} = require('../virtualization-guests-list-refresh-api');
+const React = require('react');
+const MessageContainer = require('components/messages').Messages;
+const MessagesUtils = require('components/messages').Utils;
+const { Table } = require('components/table');
+const { Column } = require('components/table');
+const { SearchField } = require('components/table');
+const Functions = require('utils/functions');
+const { LinkButton, AsyncButton } = require('components/buttons');
+const { DangerDialog } = require('components/dialog/DangerDialog');
+const { ModalButton } = require('components/dialog/ModalButton');
+const Systems = require('components/systems');
+const { VirtualizationGuestActionApi } = require('../virtualization-guest-action-api');
+const { VirtualizationGuestsListRefreshApi } = require('../virtualization-guests-list-refresh-api');
 
-class GuestsList extends React.Component {
+const { Utils } = Functions;
 
-  constructor(props) {
-    super(props);
+declare function t(msg: string, ...args: Array<any>): string;
+declare var userPrefPageSize: number;
 
-    this.state = {
-      actionsResults: {},
-      errors: [],
-      selectedItems: []
-    }
+type Props = {
+  serverId: string,
+  refreshInterval: number,
+  saltEntitled: boolean,
+  isAdmin: boolean,
+};
 
-    this.onBeforeUnload = (event) => {
-        this.setState({
-            pageUnloading: true
-        });
-    };
+type State = {
+  actionsResults: Object,
+  errors: Array<string>,
+  selectedItems: Array<Object>,
+  selected?: Object,
+  pageUnloading: boolean,
+  websocketErr: boolean,
+};
 
-    this.handleSelectItems = (items) => {
-      this.setState({
-        selectedItems: items
-      });
+class GuestsList extends React.Component<Props, State> {
+  static sortByUpdate(aRaw: Object, bRaw: Object, columnKey: string, sortDirection: number) {
```

Comment:
I wonder if these static functions wouldn't be better inside some kind of local utils file, so the component GuestList gets a bit less cumbersome. It's just a remark for future :)